MSAL/test/integration/native_auth/end_to_end/sign_up/MSALNativeAuthSignUpUsernameAndPasswordEndToEndTests.swift

```diff
@@ -464,7 +582,116 @@ final class MSALNativeAuthSignUpUsernameAndPasswordEndToEndTests: MSALNativeAuth
         await fulfillment(of: [signUpCompleteExp])
         XCTAssertTrue(signUpVerifyCodeDelegate.onSignUpCompletedCalled)
     }
-
+    
+    // Use case 1.1.10. Sign up - with Email & Password, User already exists with given email as email-pw account
+    func test_signUpWithEmailPassword_andAgainSameEmail_fails() async throws {
+        guard let sut = initialisePublicClientApplication(clientIdType: .password), let username = retrieveUsernameForSignInUsernameAndPassword() else {
+            XCTFail("Missing information")
+            return
+        }
+        
+        let password = generateRandomPassword()
+        
+        let signUpFailureExp = expectation(description: "sign-up with existing email fails")
+        let signUpStartDelegate = SignUpPasswordStartDelegateSpy(expectation: signUpFailureExp)
+        
+        sut.signUp(
+            username: username,
+            password: password,
+            correlationId: correlationId,
+            delegate: signUpStartDelegate
+        )
+        
+        await fulfillment(of: [signUpFailureExp])
+        
+        // Verify error condition
+        XCTAssertTrue(signUpStartDelegate.onSignUpPasswordErrorCalled)
+        XCTAssertEqual(signUpStartDelegate.error?.isUserAlreadyExists, true)
+    }
+    
+    // Use case 1.1.11. Sign up - with Email & Password, User already exists with given email as social account
+    func test_signUpWithEmailPassword_socialAccount_fails() async throws {
+        throw XCTSkip("Skipping test as it requires a Social account, not present in MSIDLAB")
+        
+        guard let sut = initialisePublicClientApplication() else {
+            XCTFail("Missing information")
+            return
+        }
+        
+        let username = "social_account"
+        let password = generateRandomPassword()
+        
+        let signUpFailureExp = expectation(description: "sign-up with social account email fails")
+        let signUpStartDelegate = SignUpPasswordStartDelegateSpy(expectation: signUpFailureExp)
+        
+        sut.signUp(
+            username: username,
+            password: password,
+            correlationId: correlationId,
+            delegate: signUpStartDelegate
+        )
+        
+        await fulfillment(of: [signUpFailureExp])
+        
+        // Verify error condition
+        XCTAssertTrue(signUpStartDelegate.onSignUpPasswordErrorCalled)
+        XCTAssertEqual(signUpStartDelegate.error!.isInvalidUsername, true)
```

Comment:
Try to never use `!` to force unwrap an optional
```suggestion
        XCTAssertEqual(signUpStartDelegate.error?.isInvalidUsername, true)
```

Reply:
Good point. Done and modify and check all the sign up cases. `error!.isBrowserRequired` needs to use ! instead of ?. The others use ?.